app.js

```diff
@@ -1,219 +0,0 @@
-const bodyParser = require("body-parser");
-const cookieParser = require("cookie-parser");
-const express = require("express");
-const favicon = require("serve-favicon");
-const hbs = require("hbs");
-const mongoose = require("mongoose");
-const logger = require("morgan");
-const path = require("path");
-const session = require("express-session");
-const MongoStore = require("connect-mongo");
-const bcrypt = require("bcrypt");
-const passport = require("passport");
-const LocalStrategy = require("passport-local").Strategy;
-const flash = require("connect-flash");
-const GoogleStrategy = require("passport-google-oauth20").Strategy;
-
-require("dotenv").config();
-
-// MONGOOSE SETUP
-
-const app_name = require("./package.json").name;
-const debug = require("debug")(`${app_name}:${path.basename(__filename).split(".")[0]}`);
-
-// MODEL SETUP
-
-const User = require("./models/user.js");
-const Review = require("./models/review.js");
-const Company = require("./models/company.js");
-
-// INITIALIZE EXPRESS
-
-const app = express();
-
-// ESTABLISH DB CONNECTION
-
-const clientP = mongoose
-  .connect(process.env.MONGODB_URI, {
-    useNewUrlParser: true,
-    useUnifiedTopology: true,
-    useFindAndModify: false,
-    useCreateIndex: true,
-  })
-  .then((m) => {
-    console.log(`Connected to Mongo! Database name: "${m.connections[0].name}"`);
-    return m.connection.getClient();
-  })
-  .catch((err) => {
-    console.error("Error connecting to mongo", err);
-  });
-
-// SETTING UP SESSION AND COOKIES
-
-app.use(
-  session({
-    secret: process.env.SESSION_SECRET,
-    resave: false,
-    saveUninitialized: false,
-    cookie: {
-      httpOnly: true,
-      maxAge: 10060000,
-    },
-    store: MongoStore.create({
-      clientPromise: clientP,
-      dbName: "prod",
-      stringify: false,
-      autoRemove: "interval",
-      autoRemoveInterval: 1,
-    }),
-  })
-);
-
-// MIDDLEWARE SETUP
-
-// Displaying errors and success messages. They're attached to the req object and passed to the views.
-
-app.use(function (req, res, next) {
-  const err = req.session.error;
-  const msg = req.session.notice;
-  const success = req.session.success;
-  delete req.session.error;
-  delete req.session.success;
-  delete req.session.notice;
-  if (err) res.locals.error = err;
-  if (msg) res.locals.notice = msg;
-  if (success) res.locals.success = success;
-  next();
-});
-
-// PASSPORT SET UP
-
-app.use(passport.initialize());
```

Comment:
All this whole passport logic should be abstracted out to a dedicated passport file and imported into `app.js`. Passport logic is focused on auth, our `app.js` is focused on initialising our server, we should separate our concerns